dask/array/core.py

```diff
@@ -2596,7 +2558,70 @@ def reshape(array, shape):
 
     if np.prod(shape) != array.size:
```

Comment:
Related to https://github.com/dask/dask/issues/1479. Looks like you'll need to fix the calls to `np.prod` in reshape as well so as to not overflow in size. Could either fix here, or merge, and I'll fix in https://github.com/dask/dask/pull/1478 with the rest of the changes needed.